addons/l10n_in/tests/test_tds_tcs_alert.py

```diff
@@ -0,0 +1,344 @@
+            'country_id': country_in_id,
+        })
+        cls.partner_b.write({
+            'vat': '24DJMPM8965E1ZE',
+            'l10n_in_gst_treatment': 'composition',
+            'state_id': cls.env.ref("base.state_in_gj"),
+            'country_id': country_in_id,
+        })
+
+        # ==== Company ====
+        cls.company_data["company"].write({
+            "vat": "24AAGCC7144L6ZJ",
+            "state_id": cls.env.ref("base.state_in_gj").id,
+            "street": "street1",
+            "city": "city1",
+            "zip": "123456",
+            "country_id": country_in_id,
+            'child_ids': [
+                Command.create({'name': 'Branch A'}),
+                Command.create({'name': 'Branch B'}),
+                Command.create({'name': 'Branch C'}),
+            ],
+        })
+        cls.cr.precommit.run()
+
+        cls.branch_a, cls.branch_b, cls.branch_c = cls.env.company.child_ids
+
+    def test_tcs_tds_warning(self):
+        '''
+        Test that if the per transaction limit is not exceeded,
+        the alert should not be set.
+        '''
+        move = self.env['account.move'].create({
+            'partner_id': self.partner_a.id,
+            'move_type': 'in_invoice',
+            'invoice_date': fields.Date.today(),
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': self.product_a.id,
+                    'quantity': 1,
+                    'account_id': self.purchase_account.id,
+                    'price_unit': 1000,
+                }),
+            ],
+        })
+        move.action_post()
+        self.assertEqual(move.l10n_in_tcs_tds_warning, False)
+
+    def test_tcs_tds_warning_on_exceeded_per_transaction_limit(self):
+        '''
+        Test that if the per transaction limit is exceeded,
+        the warning message should be set accordingly.
+        '''
+        move = self.env['account.move'].create({
+            'partner_id': self.partner_a.id,
+            'move_type': 'in_invoice',
+            'invoice_date': fields.Date.today(),
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': self.product_a.id,
+                    'quantity': 40,
+                    'account_id': self.purchase_account.id,
+                    'price_unit': 1000,
+                }),
+            ],
+        })
+        move.action_post()
+
+        # Check if the warning is correctly set
+        self.assertEqual(move.l10n_in_tcs_tds_warning, "It's advisable to deduct TDS u/s  192A on this transaction.")
+
+    def test_tcs_tds_warning_on_exceeded_aggregate_limit(self):
+        '''
+        Test that if the aggregate limit is exceeded,
+        the warning message should be set accordingly.
+        '''
+        move = self.env['account.move'].create({
+            'partner_id': self.partner_a.id,
+            'move_type': 'in_invoice',
+            'invoice_date': fields.Date.today(),
```

Comment:
You can use `freeze_time` decorator, using `Date.today()` is not a good practice for test cases